src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -218,6 +219,13 @@ public void Initialize()
             {
                 _deadline = DateTime.MaxValue;
             }
+
+            if (HttpContext.Request.Headers.TryGetValue(GrpcProtocolConstants.MessageEncodingHeader, out var values))
+            {
+                _grpcEncoding = values;
+            }
+
+            HttpContext.Response.Headers.Append(GrpcProtocolConstants.MessageEncodingHeader, GetGrpcEncoding() ?? GrpcProtocolConstants.IdentityGrpcEncoding);
```

Comment:
So by default the response will be encoded in the same way as the request? Hmm though esoteric, I think we should check if an grpc-accept-encoding is available on the request for us to consider. If not, we can fallback to using the same encoding as the request.

Reply:
I don't think the server should respond with the same compression as that client used. Server should not use compression unless it has been configured to do so (see e-mail for more discussion).